care/facility/api/serializers/patient.py

```diff
@@ -491,6 +501,38 @@ class PatientNotesSerializer(serializers.ModelSerializer):
         allow_null=True,
     )
     reply_to_object = ReplyToPatientNoteSerializer(source="reply_to", read_only=True)
+    files = serializers.SerializerMethodField()
+    replies = ReplyToPatientNoteSerializer(many=True, read_only=True)
+    parent_note_object = serializers.SerializerMethodField()
+    mentioned_users = serializers.SerializerMethodField()
+
+    def get_mentioned_users(self, obj):
+        mentioned_users = set(re.findall(r"@(\w+)", obj.note))
+        users = User.objects.filter(username__in=mentioned_users)
+        return UserBaseMinimumSerializer(users, many=True).data
+
+    def get_parent_note_object(self, obj):
+        parent_note = obj
+        while parent_note.reply_to is not None:
+            parent_note = parent_note.reply_to
+        return (
+            ReplyToPatientNoteSerializer(parent_note).data
+            if parent_note != obj
+            else None
+        )
```

Comment:
This recursively tries to find the root parent instead of the immediate parent right? Correct me if I'm wrong. If so why do we need it?

Reply:
we need it to get actual parent for thread view

Reply:
not sure if this is the right way to do it. wouldn't it cause too many DB queries?

cc: @sainak @vigneshhari

Reply:
If root note is required, then it is ideal to add root note to each of the reply, along with their direct parent note.

Reply:
added a new field parent_note which will set to reply_to if no parent present for reply_to or else it will set to the parent note of reply to